django/thunderstore/social/api/experimental/views/complete_login.py

```diff
@@ -0,0 +1,67 @@
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import serializers
+from rest_framework.response import Response
+from rest_framework.status import HTTP_400_BAD_REQUEST
+from rest_framework.views import APIView
+
+from thunderstore.social.providers import DiscordOauthHelper, GitHubOauthHelper
+
+
+class RequestBody(serializers.Serializer):
+    code = serializers.CharField(
+        label="Temporary code received from provider when authentication flow was initiated on client"
+    )
+    redirect_url = serializers.CharField(
+        label="Redirect URL used when the authentication flow was initiated on client"
+    )
+    secret = serializers.CharField(
+        label="Token shared between Thunderstore services to authenticate OAuth requests"
+    )
+
+
+class ResponseBody(serializers.Serializer):
+    session_id = serializers.CharField()
+
+
+class CompleteLoginApiView(APIView):
+    """
+    Complete OAuth login process initiated by a client.
+    """
+
+    permission_classes = []
+
+    @swagger_auto_schema(
+        operation_id="experimental.auth.complete",
+        request_body=RequestBody,
+        responses={200: ResponseBody()},
+    )
+    def post(self, request: HttpRequest, provider: str) -> HttpResponse:
+        secret = request.data.get("secret")
```

Comment:
Now that I've mentioned pydantic, it makes me sad DRF doesn't use a similar mechanism and we need to deal with non-typed data. We could add a duplicate declaration of the types as a pydantic model & deserialize the data into that from `request.data` to get strongly typed & validated variables though, but that might be a bit heavy on the boilerplate. I'd personally do either that or manually declare the variable types here individually (as they are validated after all).

Perhaps something exists or could be created that converts Pydantic models into DRF (de)serializers? That's out of scope for this PR however.

Reply:
Declared the types manually, seemed the easiest approach and I don't expect scaling issues here.